xdem/coreg/base.py

```diff
@@ -2092,6 +2100,14 @@ def fit(
         if self._meta["inputs"]["random"]["subsample"] != 1:
             self._meta["inputs"]["random"]["random_state"] = random_state
 
+        # Apply the shift to the source dem if given
+        initial_shift_apply = False
+        if self._meta["inputs"]["affine"].get("initial_shift") is not None:
+            shift_x = self._meta["inputs"]["affine"]["initial_shift"][0]  # type: ignore
+            shift_y = self._meta["inputs"]["affine"]["initial_shift"][1]  # type: ignore
+            reference_elev = reference_elev.translate(-shift_x, -shift_y)  # type: ignore
+            initial_shift_apply = True
```

Comment:
This applies X/Y but no Z shift? 
Not sure if the idea is to also apply Z shift, or allow only a `tuple[float, float]`. Both work for me!